spec/reek/smells/too_many_constants_spec.rb

```diff
@@ -0,0 +1,125 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/too_many_constants'
+require_relative 'smell_detector_shared'
+
+RSpec.describe Reek::Smells::TooManyConstants do
+  let(:detector) { described_class.new(max_constants: default_max_consts) }
+
+  it_should_behave_like 'SmellDetector'
+
+  def default_max_consts
+    described_class::DEFAULT_MAX_CONSTS
+  end
+
+  def too_many_consts
+    default_max_consts + 1
+  end
+
+  def casgn_sequence(count: default_max_consts, consts: ('A'..'Z').to_a)
+    consts.first(count).map.with_index { |name, i| "#{name}=#{i}" }.join("\n")
+  end
+
+  context 'counting constants' do
+    it 'should not report for non-excessive constants' do
+      src = <<-EOS
+        class Dummy
+          #{casgn_sequence}
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should count each constant only once' do
+      src = <<-EOS
+        class Good
+          #{casgn_sequence}
+        end
+
+        class Bad
+          #{casgn_sequence}
+        end
+
+        class Ugly
+          #{casgn_sequence}
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should report for excessive constants inside classes' do
+      src = <<-EOS
+        class Dummy
+          #{casgn_sequence(count: too_many_consts)}
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants)
+    end
+
+    xit 'should report for excessive constants anywhere' do
+      src = <<-EOS
+        #{casgn_sequence(count: too_many_consts)}
+
+        module Dummiest
+          class Dummy
+          end
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants)
+    end
+
+    it 'should report for excessive constants inside modules' do
+      src = <<-EOS
+        module Dummiest
+          #{casgn_sequence(count: too_many_consts)}
+
+          class Dummy
+          end
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants)
+    end
+  end
+
+  context 'smell report' do
```

Comment:
Alternatively, you can at least pull the definition of `src` out into a `let` statement.